handlers/metrics.go

```diff
@@ -376,7 +376,7 @@ func prepareStatsQueries(w http.ResponseWriter, r *http.Request, rawQ []models.M
 			}
 		}
 		if !found {
-			newNs := models.Namespace{Name: q.Target.Namespace, Cluster: q.Target.Cluster}
+			newNs := models.Namespace{Name: q.Target.Namespace, Cluster: models.ClusterNameFromTarget(q.Target)}
```

Comment:
@jshaughn tbh I would leave `q.Validate` as a validate only and not change any value there. In the section where `models.ClusterNameFromTarget` is called, it is not using the query but preparing a namespaces list based on queries, so IMO this is a good place to put the `cluster` into Namespace when cluster is empty, as in the next call the namespaces list is used. And in the next section, as you said, the `q.Validate` is  called before using the queries.
So I think we need to leave it as it is, if I understand your question correctly?

My initial concern was to have something like `createMetricsLabelsBuilder` to set th defaults (instead of Validate), but found out that it is a huge refactoring rather that a fix for missing `cluster` param in json.

Reply:
`q.Validate` is already setting a value (`q.QueryTime`) based on a passed-in value (`q.RawQueryTime`), so I thought that this would be analogous.  But if you feel it's good this way I'm fine with it.

Not really sure we need the single-use `ClusterNameFromTarget`, could just add that logic inline, above.  Or, if you wanted a more general-purpose utility maybe add something in Config.go like `GetSafeClusterName(cluster: string) string`, that other packages may want to use as well.